### PR TITLE
Add utils_npiv python file

### PR DESCRIPTION
**Pull request by lento-sun**
This utils file is used for npiv related operations, such as:
nodedev create/destroy, mpath pool xml prepare, find online
vhbas, etc.

sign-off by: Yi Sun (lentosun@163.com)

### CODE REVIEW

File: virttest/utils_npiv.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+
+def restart_multipathd(mpath_dev="", expect_exist=False):
+    """
+    Restart the multipath deamon, and check if mpath_dev still exists
+    after deamon restarted, as expected.
+    """
+    cmd_status = process.system('service multipathd restart', verbose=True)
+    if cmd_status:
+        raise TestFail("Restart multipathd failed, please check your system.")
```
Comment (harish-24) [inline virttest/utils_npiv.py]: should be `exceptions.TestFail`

File: virttest/utils_npiv.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    :param kwargs: Could contain addr_bus, addr_target,
+     addr_unit, mode, and managed
+    :return: an xml object set by kwargs
+    """
+    addr_bus = kwargs.get('addr_bus', 0)
+    addr_target = kwargs.get('addr_target', 0)
+    addr_unit = kwargs.get('addr_unit', 0)
+    mode = kwargs.get('mode', 'subsystem')
+    managed = kwargs.get('managed', 'no')
+    fc_host_dir = kwargs.get('fc_host_dir', '/sys/class/fc_host')
```
Comment (harish-24) [inline virttest/utils_npiv.py]: `fc_host_dir` is not used

**Comment by lento-sun** [comment]
@harish-24 thx for the review, updated accordingly.

File: virttest/utils_npiv.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+
+def create_hostdev_xml(adapter_name="", **kwargs):
+    """
+    Create vhba hostdev xml.
+
+    :param adapter_name: The name of the scsi adapter
+    :param kwargs: Could contain addr_bus, addr_target,
+     addr_unit, mode, and managed
+    :return: an xml object set by kwargs
```
Comment (will-Do) [inline virttest/utils_npiv.py]: s/an/a

File: virttest/utils_npiv.py [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+    lun_dicts = []
+    tmp_list = []
+    scsi_number = scsi_host.replace("scsi_host", "")
+    cmd = "multipath -ll | grep '\- %s:' | grep 'ready running' |\
+           awk '{FS=\" \"}{for (f=1; f<=NF; f+=1) {if ($f ~ /%s:/)\
+           {print $f}}}'" % (scsi_number, scsi_number)
+    try:
+        result = process.run(cmd, shell=True)
+    except Exception, e:
+        raise exceptions.TestError("Error occurred when run cmd: multipath")
```
Comment (will-Do) [inline virttest/utils_npiv.py]: It's better to print the whole exception message

File: virttest/utils_npiv.py [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    Restart the multipath deamon, and check if mpath_dev still exists
+    after deamon restarted, as expected.
+    """
+    cmd_status = process.system('service multipathd restart', verbose=True)
+    if cmd_status:
+        raise exceptions.TestFail("Restart multipathd failed.")
+    if not os.path.exists(mpath_dev):
+        if not expect_exist:
+            return True
+    if os.path.exists(mpath_dev):
```
Comment (will-Do) [inline virttest/utils_npiv.py]: else

**Comment by lento-sun** [comment]
@will-Do Thx for review, updated accordingly.

**Comment by xutian** [comment]
@lento-sun   Pep8 check fail, please fix it and re-base your tree

**Comment by lento-sun** [comment]
@xutian Thx for the reminding, it was produced by a recent change. modified.

**Pull request merged**